File: spoj/7974-ACPC10A/solution_7974.cpp
#include <iostream>

using namespace std;

void findAPGP(int x, int y, int z);

int main()
{
    while (true)
    {
        int x, y, z;
        cin >> x >> y >> z;

        if (x == 0 && y == 0 && z == 0)
            break;

        findAPGP(x, y, z);

        cout << "\n";
    }
    return 0;
}

void findAPGP(int x, int y, int z)
{
    int diff = y - x;

    if (z - y == diff)
        cout << "AP " << z + diff;
    else
        cout << "GP " << (z * y) / x;
}

File: spoj/7974-ACPC10A/solution_7974.py
def find_ap_gp(x, y, z):
    diff = y - x
    if z - y == diff:
        print(f"AP {(z + diff)}")
    else:
        print(f"GP {int(z * y / x)}")


if __name__ == "__main__":
    while True:
        inp = input().split(" ")
        x, y, z = list(map(int, inp))
        if x == y == z == 0:
            break
        find_ap_gp(x, y, z)
